diff -up chromium-45.0.2454.101/content/browser/time_zone_monitor.cc.system-icu chromium-45.0.2454.101/content/browser/time_zone_monitor.cc
--- chromium-45.0.2454.101/content/browser/time_zone_monitor.cc.system-icu	2015-10-13 13:42:42.816642333 -0400
+++ chromium-45.0.2454.101/content/browser/time_zone_monitor.cc	2015-10-13 13:43:13.750917363 -0400
@@ -22,7 +22,7 @@ TimeZoneMonitor::~TimeZoneMonitor() {
 
 void TimeZoneMonitor::NotifyRenderers() {
   DCHECK_CURRENTLY_ON(BrowserThread::UI);
-#if defined(OS_CHROMEOS)
+#if defined(OS_CHROMEOS) || U_ICU_VERSION_MAJOR_NUM < 57
   // On CrOS, ICU's default tz is already set to a new zone. No
   // need to redetect it with detectHostTimeZone().
   scoped_ptr<icu::TimeZone> new_zone(icu::TimeZone::createDefault());
