Upstream: No
Source: Gentoo

From e1bbdec720a333937bd1b990ae0f7ee97db0d3b0 Mon Sep 17 00:00:00 2001
From: Your Name <you@example.com>
Date: Fri, 28 Jun 2019 15:56:23 +0000
Subject: [PATCH] update zlib

---
--- a/third_party/perfetto/gn/BUILD.gn.old	2020-02-13 21:37:52.314600930 +0000
+++ b/third_party/perfetto/gn/BUILD.gn	2020-02-13 21:42:25.694349607 +0000
@@ -294,7 +294,7 @@
       public_configs = [ "//buildtools:zlib_config" ]
       public_deps = [ "//buildtools:zlib" ]
     } else {
-      public_configs = [ "//third_party/zlib:zlib_config" ]
+      public_configs = [ "//third_party/zlib:system_zlib" ]
       public_deps = [ "//third_party/zlib" ]
     }
   }
