Index: chromium-65.0.3325.146/media/gpu/vaapi/vaapi_wrapper.cc
===================================================================
--- chromium-65.0.3325.146.orig/media/gpu/vaapi/vaapi_wrapper.cc
+++ chromium-65.0.3325.146/media/gpu/vaapi/vaapi_wrapper.cc
@@ -350,6 +350,8 @@ static VAEntrypoint GetVaEntryPoint(Vaap
     case VaapiWrapper::kCodecModeMax:
       NOTREACHED();
       return VAEntrypointVLD;
+    default:
+      return 0;
   }
 }
 

