diff -urN chromium-57.0.2987.98.orig/chrome/browser/ui/libgtkui/gtk_ui.cc chromium-57.0.2987.98/chrome/browser/ui/libgtkui/gtk_ui.cc
--- chromium-57.0.2987.98.orig/chrome/browser/ui/libgtkui/gtk_ui.cc	2017-03-10 12:10:00.702490379 +0300
+++ chromium-57.0.2987.98/chrome/browser/ui/libgtkui/gtk_ui.cc	2017-03-10 16:29:20.464894725 +0300
@@ -676,7 +676,7 @@
             : nullptr);
   }
 
-  return gtk_border;
+  return std::move(gtk_border);
 }
 
 void GtkUi::AddWindowButtonOrderObserver(
diff -urN chromium-57.0.2987.98.orig/remoting/host/security_key/fake_security_key_ipc_server.cc chromium-57.0.2987.98/remoting/host/security_key/fake_security_key_ipc_server.cc
--- chromium-57.0.2987.98.orig/remoting/host/security_key/fake_security_key_ipc_server.cc	2017-03-10 12:10:01.450498383 +0300
+++ chromium-57.0.2987.98/remoting/host/security_key/fake_security_key_ipc_server.cc	2017-03-10 16:30:49.818666156 +0300
@@ -130,7 +130,7 @@
 
   ipc_server_map_[connection_id] = fake_ipc_server->AsWeakPtr();
 
-  return fake_ipc_server;
+  return std::move(fake_ipc_server);
 }
 
 base::WeakPtr<FakeSecurityKeyIpcServer>
diff -urN chromium-56.0.2924.76.orig/third_party/webrtc/modules/desktop_capture/screen_capturer_x11.cc chromium-56.0.2924.76/third_party/webrtc/modules/desktop_capture/screen_capturer_x11.cc
--- chromium-56.0.2924.76.orig/third_party/webrtc/modules/desktop_capture/screen_capturer_x11.cc	2017-01-26 21:07:25.725445907 +0300
+++ chromium-56.0.2924.76/third_party/webrtc/modules/desktop_capture/screen_capturer_x11.cc	2017-01-26 21:11:22.865332765 +0300
@@ -412,7 +412,7 @@
     return nullptr;
   }
 
-  return capturer;
+  return std::move(capturer);
 }
 
 }  // namespace webrtc
