diff -urN chromium-51.0.2704.63.orig/third_party/webrtc/modules/video_coding/codecs/h264/h264_decoder_impl.cc chromium-51.0.2704.63/third_party/webrtc/modules/video_coding/codecs/h264/h264_decoder_impl.cc
--- chromium-51.0.2704.63.orig/third_party/webrtc/modules/video_coding/codecs/h264/h264_decoder_impl.cc	2016-05-26 14:23:10.796936658 +0300
+++ chromium-51.0.2704.63/third_party/webrtc/modules/video_coding/codecs/h264/h264_decoder_impl.cc	2016-05-26 14:46:46.806085065 +0300
@@ -15,9 +15,9 @@
 #include <limits>
 
 extern "C" {
-#include "third_party/ffmpeg/libavcodec/avcodec.h"
-#include "third_party/ffmpeg/libavformat/avformat.h"
-#include "third_party/ffmpeg/libavutil/imgutils.h"
+#include <libavcodec/avcodec.h>
+#include <libavformat/avformat.h>
+#include <libavutil/imgutils.h>
 }  // extern "C"
 
 #include "webrtc/base/checks.h"
diff -urN chromium-51.0.2704.63.orig/third_party/webrtc/modules/video_coding/codecs/h264/h264_decoder_impl.h chromium-51.0.2704.63/third_party/webrtc/modules/video_coding/codecs/h264/h264_decoder_impl.h
--- chromium-51.0.2704.63.orig/third_party/webrtc/modules/video_coding/codecs/h264/h264_decoder_impl.h	2016-05-26 14:23:10.794936602 +0300
+++ chromium-51.0.2704.63/third_party/webrtc/modules/video_coding/codecs/h264/h264_decoder_impl.h	2016-05-26 14:46:15.335133108 +0300
@@ -17,7 +17,7 @@
 #include "webrtc/modules/video_coding/codecs/h264/include/h264.h"
 
 extern "C" {
-#include "third_party/ffmpeg/libavcodec/avcodec.h"
+#include <libavcodec/avcodec.h>
 }  // extern "C"
 
 #include "webrtc/common_video/include/i420_buffer_pool.h"
