Index: tools/mb/mb_config.pyl
diff --git a/tools/mb/mb_config.pyl b/tools/mb/mb_config.pyl
index 1aee239cebb807c4cd082c33beb6ea2b886072cb..29aae712e683949e8a96628c43e286e4b76a0c7e 100644
--- a/tools/mb/mb_config.pyl
+++ b/tools/mb/mb_config.pyl
@@ -1582,7 +1582,7 @@
     },
 
     'clang_tot': {
-      'gn_args': 'llvm_force_head_revision=true clang_use_chrome_plugins=false',
+      'gn_args': 'llvm_force_head_revision=true clang_use_chrome_plugins=false enable_nocompile_tests=false',
       'mixins': ['clang'],
     },
 
