--- a/build/toolchain/gcc_toolchain.gni	2018-07-24 16:08:13.000000000 -0600
+++ b/build/toolchain/new.gni	2018-07-29 19:29:45.046559458 -0600
@@ -339,12 +339,12 @@
         # We use slightly different arflags for AIX.
         extra_arflags = "-r -c -s"
       } else {
-        extra_arflags = "-r -c -s -D"
+        extra_arflags = "rcsD"
       }
 
       # Almost all targets build with //build/config/compiler:thin_archive which
       # adds -T to arflags.
-      command = "$python_path \"$ar_wrapper\"$whitelist_flag --output={{output}} --ar=\"$ar\" \"{{arflags}} $extra_arflags\" @\"$rspfile\""
+      command = "$python_path \"$ar_wrapper\"$whitelist_flag --output={{output}} --ar=\"$ar\" \"$extra_arflags{{arflags}}\" @\"$rspfile\""
       description = "AR {{output}}"
       rspfile_content = "{{inputs}}"
       outputs = [
--- a/build/config/compiler/BUILD.gn	2018-07-24 16:08:13.000000000 -0600
+++ b/build/config/compiler/BUILD-new.gn	2018-07-29 19:32:54.113525614 -0600
@@ -1595,7 +1595,7 @@
   # have a "thin archive" mode (it does accept -T, but it means truncating
   # archive names to 16 characters, which is not what we want).
   if ((is_posix && !is_nacl && !is_mac && !is_ios) || is_fuchsia) {
-    arflags = [ "-T" ]
+      arflags = [ "T" ]
   }
 }
 
